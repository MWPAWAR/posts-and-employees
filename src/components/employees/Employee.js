import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

const Employee = ({
  id,
  address: { suite, street, city, zipcode },
  name,
}) => (
  <Card className="root">
    <CardActionArea>
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">{name}</Typography>
        <Typography variant="body1" color="textSecondary" component="p">
          <FormattedMessage
            id="app.employees.address"
            values={{ suite, street, city, zipcode }}
          />
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Link to={`/employees/${id}`}>
        <Button size="large" color="primary">
          <FormattedMessage id="app.employees.btn-view-label" />
        </Button>
      </Link>
    </CardActions>
  </Card>
);

Employee.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.shape({
    street: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    suite: PropTypes.string.isRequired,
    zipcode: PropTypes.string.isRequired
  }).isRequired
};

export default Employee;
