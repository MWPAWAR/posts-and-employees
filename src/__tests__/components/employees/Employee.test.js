import React from 'react';
import { shallow } from 'enzyme';

import Employee from '../../../components/employees/Employee';
import employeeMock from '../../mocks/employee.json';

describe('Employee', () => {
  test('it renders correctly with default data', () => {
    const component = shallow(<Employee {...employeeMock} />);
    expect(component).toMatchSnapshot();
  });
});
