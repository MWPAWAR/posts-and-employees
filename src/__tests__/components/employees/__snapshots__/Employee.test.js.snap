// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee it renders correctly with default data 1`] = `
<WithStyles(ForwardRef(Card))
  className="root"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        Glenna Reichert
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body1"
      >
        <FormattedMessage
          id="app.employees.address"
          values={
            Object {
              "city": "Bartholomebury",
              "street": "Dayna Park",
              "suite": "Suite 449",
              "zipcode": "76495-3109",
            }
          }
        />
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <Link
      to="/employees/9"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="large"
      >
        <FormattedMessage
          id="app.employees.btn-view-label"
        />
      </WithStyles(ForwardRef(Button))>
    </Link>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
